File: src/main/scala/com/tomergabel/examples/patterns/PatternMatching2.scala
package com.tomergabel.examples.patterns

/**
  * Created by tomerga on 04/04/2016.
  */
object PatternMatching2 extends App {

  val Url = "(\\w+)://([a-zA-Z0-9.-]+?)(/[a-zA-Z0-9.-/]*)?".r

  args.head match {
    case Url(scheme, host, path) => println(s"Scheme: $scheme\nHost: $host\nPath: $path")
    case arg => println(s"Not a recognized URL: $arg")
  }
}

File: src/main/java/com/tomergabel/examples/java/PatternMatching2.java
package com.tomergabel.examples.java;

import java.util.regex.Matcher;
import java.util.regex.Pattern;

/**
 * Created by tomerga on 05/04/2016.
 */
public class PatternMatching2 {

    public static void main(String[] args) {
        Pattern url = Pattern.compile("(\\w+)://([a-zA-Z0-9.-]+?)(/[a-zA-Z0-9.-/]*)?");
        Matcher matcher = url.matcher(args[0]);
        if (matcher.matches()) {
            String scheme = matcher.group(1);
            String host = matcher.group(2);
            String path = matcher.group(3);
            System.out.println("Scheme: " + scheme + "\nHost: " + host + "\nPath: " + path);
        } else
            System.out.println("Not a recognized URL: " + args[0]);
    }
}
